DiamondLightSource/python-artemis
# 760 refactor total count optimisation tests

Fixes #760 

Draft PR until https://github.com/DiamondLightSource/python-artemis/pull/755 is merged

### To test:
1. Run tests
2. Check new tests test all the same logic as the removed "test_total_counts_iteration"
